score.go

```diff
@@ -130,7 +152,19 @@ func WithPeerScoreInspect(inspect PeerScoreInspectFn, period time.Duration) Opti
 			return fmt.Errorf("peer scoring is not enabled")
 		}
 
-		gs.score.inspect = inspect
+		switch i := inspect.(type) {
+		case PeerScoreInspectFn:
+			gs.score.inspect = i
+		case func(map[peer.ID]float64):
```

Comment:
Alternatively you could change these to type aliases, then you wouldn't need the explicit type-casting.

Reply:
done, made them type aliases.